templates/components/api-server/template/server/server.js

```diff
@@ -3,19 +3,21 @@ var boot = require('loopback-boot');
 
 var app = module.exports = loopback();
 
-// Bootstrap the application, configure models, datasources and middleware.
-// Sub-apps like REST API are mounted via boot scripts.
-boot(app, __dirname);
-
 app.start = function() {
   // start the web server
   return app.listen(function() {
-    app.emit('started');
     console.log('Web server listening at: %s', app.get('url'));
   });
 };
 
-// start the server if `$ node server.js`
-if (require.main === module) {
-  app.start();
-}
+// Bootstrap the application, configure models, datasources and middleware.
+// Sub-apps like REST API are mounted via boot scripts.
+boot(app, __dirname, function(err) {
+  if (err) throw err;
+
+  app.emit('ready');
```

Comment:
An earlier comment says "This should be implemented by loopback-boot, please remove". Boot already emits `booted` event, this is what consumer should use.